lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1232,12 +1231,40 @@ private synchronized void onConnected(ProtocolMessage message) {
             requestState(transport, new StateIndication(ConnectionState.failed, e.errorInfo));
             return;
         }
-
         /* indicated connected currentState */
         setSuspendTime();
         requestState(new StateIndication(ConnectionState.connected, error));
     }
 
+    /**
+     * Add all pending queued messages to the front of QueuedMessages for them to be sent later
+     * Spec: RTN19a
+     * @param resetMessageSerial whether to reset message serial, this will determine whether to reset message serials
+     * on pending queue, for example when a connection resume failed
+     */
+    private void addPendingMessagesToQueuedMessages(boolean resetMessageSerial) {
+        // Add messages from pending messages to front of queuedMessages in order to retry them
+        queuedMessages.addAll(0, pendingMessages.queue);
+        //Clear the pendingMessages queue now, because we do not want the retried messages to accumulate on it.
+        pendingMessages.clearQueue();
+
+        //RTN19a
+        if (resetMessageSerial){
+            pendingMessages.resetStartSerial(0);
+            // reassign new serials for queued messages - we can set them to their indexes since they
+            // will also start with 0
+            for (int i = 0; i < queuedMessages.size(); i++) {
```

Comment:
since the `msgSerial` is set later in `sendImpl()`, I don't think it's necessary to set it here.

Reply:
d8c21269f39b89b6ec294d39752b4d5b61765ab7